### DISCUSSION TITLE
Searching for wifi audio streaming client-server esp32 INMP441 (mic) / MAX98357  (amplifier) arduino examples

### DISCUSSION THREAD

**Discussion by DragonZideutch** [pschatzmann/arduino-audio-driver#36] [Q&A] [answered]
Tried differents arduino sketches, but it does not give enough stability, since AUDIOTOOLS seems to be my best bet, i tried some examples i2s project with serial monitor (works), tried some tcp server-client sinewave (works too), but i can't seem to get the best options for what i want to do. A working example would point me in the right direction i guess. What are my options. I learn by looking at working code so i can understand my error and why it failed/worked. Thanks for your help!

**Comment by pschatzmann** [comment, marked as answer]
Try to read the Wiki and examples of the [AudioTools Project](https://github.com/pschatzmann/arduino-audio-tools) first, so that you understand how to use this library.
It is true that I made most examples for the AudioKit because it is just easier to test, but usually you can just exchange the source or destination with an I2SStream or an other input or output class.

Both your input is I2S and your output is I2S, so you are looking for the stream-i2s-i2s examples. There are two of them one using a single I2S port in douplex and one using two I2S ports.

I you read the ADC Wiki, you will find the restrictions for the INMP441... 

Or maybe I did not quite understand what you try to do  ?

PS. I just noticed that you are completly wrong here: This is the project that provides the drivers for different codec chips, one of them is the es8088 for the AudioKit...

**Comment by DragonZideutch** [comment]
thanks a lot, sometimes when you have so many options a person like me gets confused. but pointing me in the right direction helps me narrowing the solutions. i am a ham radio guy and all i want is to be able to stream the audio from a remote shack to my home in town. i will get to the solution somehow, but it is great what you have done with AUDIOTOOLS , it makes it much simpler.thanks again from canada, i will try to send you the final project video once it is done. how about buying you a coffee? do you have a way for me doing a donation?

**Comment by pschatzmann** [comment]
There is a communications wiki with plenty of examles.
The sponsorship information can be found in the Readme of the [AudioTools](https://github.com/pschatzmann/arduino-audio-tools)
